addons/purchase_stock/models/stock.py

```diff
@@ -114,6 +115,11 @@ def _compute_visibility_days(self):
                 orderpoint.visibility_days = orderpoint.purchase_visibility_days
         return res
 
+    @api.depends('product_tmpl_id', 'product_tmpl_id.seller_ids', 'product_tmpl_id.seller_ids.sequence', 'product_tmpl_id.seller_ids.partner_id')
+    def _compute_product_supplier_id(self):
+        for orderpoint in self:
+            orderpoint.product_supplier_id = orderpoint.product_tmpl_id.seller_ids.sorted('sequence')[0].partner_id.id if orderpoint.product_tmpl_id.seller_ids else False
```

Comment:
Purely visual suggestion but instead of a single long line, would you mind to split it to make it easier to read ?
For example:
```suggestion
        self.product_supplier_id = False
        for orderpoint in self:
            if orderpoint.product_tmpl_id.seller_ids:
                orderpoint.product_supplier_id = orderpoint.product_tmpl_id.seller_ids.sorted('sequence')[0].partner_id.id
```

Or if you want to keep the code short (`id` of empty recordset is `False`):
```suggestion
        for orderpoint in self:
            orderpoint.product_supplier_id = orderpoint.product_tmpl_id.seller_ids.sorted('sequence')[:1].partner_id.id
```

Reply:
Sure